crates/sui-graphql-rpc/src/main.rs

```diff
@@ -96,9 +100,24 @@ async fn main() {
                 ..ServerConfig::default()
             };
 
-            start_graphiql_server(&server_config, &VERSION)
-                .await
-                .unwrap();
+            let cancellation_token_clone = cancellation_token.clone();
+            tracker.spawn(async move {
+                start_graphiql_server(&server_config, &VERSION, cancellation_token_clone)
+                    .await
+                    .unwrap();
+            });
+
+            // Wait for shutdown signal
+            tokio::select! {
+                _ = tokio::signal::ctrl_c() => {},
+            }
```

Comment:
you probably also want to select on the service itself, in the event that there's an error or it crashes you want the system to close

Reply:
Thanks for calling this out! Updated and verified with:
```
            let graphql_service_handle = tracker.spawn(async move {
                tokio::select! {
                    _ = start_graphiql_server(&server_config, &VERSION, cancellation_token_clone) => {
                        // If the server stops on its own, we'll end up here.
                        println!("GraphQL service exited on its own.");
                    }
                    _ = tokio::time::sleep(tokio::time::Duration::from_secs(10)) => {
                        // After 10 seconds, this branch will be taken,
                        // effectively "cancelling" the above task if it's still running.
                        println!("10 seconds elapsed, stopping GraphQL service.");
                    }
                }
            });

            // Wait for shutdown signal
            tokio::select! {
                result = graphql_service_handle => {
                    if let Err(e) = result {
                        println!("GraphQL service crashed or exited with error: {:?}", e);
                    }
                }
                _ = tokio::signal::ctrl_c() => {
                    println!("Ctrl+C signal received.");
                },
            }

            println!("Shutting down...");

            // Send shutdown signal to application
            cancellation_token.cancel();
            tracker.close();
            tracker.wait().await;
```